Scripts/Runtime/Jobs/Deferred/JobDeferredNativeArrayForBatchExtension.cs

```diff
@@ -111,10 +116,20 @@ internal static void Initialize()
 #if ENABLE_UNITY_COLLECTIONS_CHECKS
                     JobsUtility.PatchBufferMinMaxRanges(bufferRangePatchData, UnsafeUtility.AddressOf(ref jobData), beginIndex, endIndex - beginIndex);
 #endif
-
                     jobData.Execute(beginIndex, endIndex - beginIndex);
                 }
             }
+
+            private const int DEFAULT_NATIVE_THREAD_INDEX = -1;
+
+            private TJob m_JobData;
+            [NativeSetThreadIndex] private readonly int m_NativeThreadIndex;
+            
+            public JobDeferredNativeArrayForBatchProducer(ref TJob jobData)
+            {
+                m_JobData = jobData;
+                m_NativeThreadIndex = DEFAULT_NATIVE_THREAD_INDEX;
+            }
```

Comment:
What was this struct used for before? Does it make sense that we're tacking this functionality onto the producer?

It looks like `JobDeferredNativeArrayForBatchProducer` was just a struct with a bunch of static methods?